contrib/epee/include/string_tools.h

```diff
@@ -188,7 +188,8 @@ POP_WARNINGS
 		return boost::lexical_cast<std::string>(val);
 	}
 	//----------------------------------------------------------------------------
-	inline std::string to_string_hex(uint32_t val)
+	template<typename T>
+	inline std::string to_string_hex(const T &val)
 	{
 		std::stringstream ss;
```

Comment:
```c++
static_assert(std::is_arithmetic<T>::value, "only arithmetic types")
```
might be useful, otherwise its possibly ignoring `std::hex` attribute.